### ISSUE TITLE
802. Find Eventual Safe States: Minor inaccuracy in problem description.

### ISSUE THREAD

**Issue by kramasamy90** [LeetCode-Feedback/LeetCode-Feedback#7926]
#### Category of the bug
- [x] Question
- [ ] Solution
- [x] Language
- [ ] Missing Test Cases 

#### Description of the bug

tl;dr: I think the definition of safe node should use **"every possible maximal path"** instead of just **"every possible path"**.

**Explanation**:
Consider the following example as input:
[[1,2],[0],[3],[4],[]]

The expected output would be 
LeetCode's "Expected" output is `[2,3,4]`. 
According to your definition 2 is not a safe node. Therefore the output should be `[3,4]`. The reason is as follows.
If you look at all the paths starting from 2 we have the following two path.

1. 2 -> 3.
2. 2 -> 3 -> 4.

Among these only the second path ends in a terminal node. Therefore 2 is not a safe node according the current problem description.

But, if you change the definition of safe node to "every possible **maximal** path should end in a terminal node", there will be only one maximal path starting form 2, which is the second one. Thus, 2 would be a safe node and the expected output would match the problem description.

**Comment by LC-Pam** [comment]
Thank you for your time. We've used your feedback to update the problem.  We appreciate your support!
